frontend/public/components/sidebars/resource-sidebar-samples.tsx

```diff
@@ -0,0 +1,284 @@
+          'Limit the size and number of persistent volume claims that can be created with a storage class.',
+        templateName: 'rq-storageclass',
+        kind: referenceForModel(ResourceQuotaModel),
+      },
+    ],
+  )
+  .setIn(
+    [referenceForModel(RoleModel)],
+    [
+      {
+        title: 'Allow reading the resource in API group',
+        description: 'This "Role" is allowed to read the resource "Pods" in the core API group.',
+        templateName: 'read-pods-within-ns',
+        kind: referenceForModel(RoleModel),
+      },
+      {
+        title: 'Allow reading/writing the resource in API group',
+        description:
+          'This "Role" is allowed to read and write the "Deployments" in both the "extensions" and "apps" API groups.',
+        templateName: 'read-write-deployment-in-ext-and-apps-apis',
+        kind: referenceForModel(RoleModel),
+      },
+      {
+        title: 'Allow different access rights to different types of resource and API groups',
+        description:
+          'This "Role" is allowed to read "Pods" and read/write "Jobs" resources in API groups.',
+        templateName: 'read-pods-and-read-write-jobs',
+        kind: referenceForModel(RoleModel),
+      },
+      {
+        title: 'Allow reading a ConfigMap in a specific namespace (for RoleBinding)',
+        description:
+          'This "Role" is allowed to read a "ConfigMap" named "my-config" (must be bound with a "RoleBinding" to limit to a single "ConfigMap" in a single namespace).',
+        templateName: 'read-configmap-within-ns',
+        kind: referenceForModel(RoleModel),
+      },
+      clusterRoleBinding1,
+      clusterRoleBinding2,
+    ],
+  )
+  .setIn([referenceForModel(ClusterRoleModel)], [clusterRoleBinding1, clusterRoleBinding2]);
+
+export const getResourceSidebarSamples = (kindObj: K8sKind, yamlSampleObj: K8sResourceKind) => {
+  const yamlSampleData = !_.isEmpty(yamlSampleObj)
+    ? _.filter(
+        yamlSampleObj.data,
+        (sample) => sample.spec.apiVersion === apiVersionForModel(kindObj),
+      )
+    : [];
+  const yamlSamples = [];
+  _.each(yamlSampleData, (sample) => {
+    yamlSamples.push(_.get(sample, 'spec'));
+  });
+  const existingSamples = defaultSamples.get(referenceForModel(kindObj));
+  return !_.isEmpty(existingSamples) ? existingSamples.concat(yamlSamples) : yamlSamples;
+};
+
+const ResourceSidebarSample: React.FC<ResourceSidebarSampleProps> = ({
+  sample,
+  loadSampleYaml,
+  downloadSampleYaml,
+}) => {
+  const { highlightText, title, img, description, templateName, yaml, kind } = sample;
+  return (
+    <li className="co-resource-sidebar-item">
+      <h3 className="h4">
+        <span className="text-uppercase">{highlightText}</span> {title}
+      </h3>
+      {img && <img src={img} className="co-resource-sidebar-item__img img-responsive" />}
+      <p>{description}</p>
+      <Button variant="link" isInline onClick={() => loadSampleYaml(templateName, yaml, kind)}>
```

Comment:
@sg00dwin:  no need to migrate these buttons (note they've been moved here from frontend/public/components/sidebars/resource-sidebar.jsx)